pygsti/layouts/matrixlayout.py

```diff
@@ -290,16 +311,57 @@ def __init__(self, circuits, model, dataset=None, num_sub_trees=None, num_tree_p
         unique_circuits, to_unique = self._compute_unique_circuits(circuits)
         aliases = circuits.op_label_aliases if isinstance(circuits, _CircuitList) else None
         ds_circuits = _lt.apply_aliases_to_circuits(unique_circuits, aliases)
-        unique_complete_circuits, split_unique_circuits = model.complete_circuits(unique_circuits, return_split=True)
+
+        #extract subcaches from layout_creation_circuit_cache:
+        if layout_creation_circuit_cache is not None:
```

Comment:
Since we are using `.get(*, None)` anyway, thoughts on including

```
if layout_creation_circuit_cache is None:
    layout_creation_circuit_cache = {}
```

above and then removing this if/else? It may also cut down on branching in other places.